### ISSUE TITLE
New dark tiding "Static Charge" upgrade card doesn't work

### ISSUE THREAD

**Issue by gian238** [keyteki/keyteki#2421]
As the title suggests, the logos upgrade card "Static Charge" doesn't work and doesn't assign damages to the neighbor creatures
![static-charge](https://user-images.githubusercontent.com/82649874/115009614-c153f300-9eac-11eb-957d-7a2c6374b27b.png)

**Comment by Lynzkar** [comment]
I had a game earlier today where it did, could it perhaps have been attached to a creature like Shoulder ID that cannot deal damage?

**Comment by gian238** [comment]
> I had a game earlier today where it did, could it perhaps have been attached to a creature like Shoulder ID that cannot deal damage?

Yes, actually it happened with the card being linked to shoulder id, however it's strange because the damages would be dealt by the upgrade rather than the creature itself.

**Comment by blinkingline** [comment]
In this case, Static Charge's ability grants a new ability to Shoulder Id, meaning Shoulder Id's text box now reads (numbers for reference):

1. Taunt.
2. Shoulder Id cannot fight.
3. When Shoulder Id would deal damage, steal 1 Æmber instead.
4. At the start of your turn, deal 2 to each of this creature's neighbors.

At the start of the Shoulder Id's controller's turn, Ability 4 resolves and Shoulder Id would deal 2 to each of the creature's neighbors. However, since Shoulder Id would deal damage, that dealing of damage is replaced with a steal via Ability 3, and Shoulder Id's controller steals 1 Æmber.
Since all damage is dealt simultaneously, the dealing of the damage only happens once, which means the replacement of that damage will also only happen once, resulting in Shoulder Id stealing 1 Æmber.

**Comment by Blaph** [comment]
> 
> 
> In this case, Static Charge's ability grants a new ability to Shoulder Id, meaning Shoulder Id's text box now reads (numbers for reference):
> 
>     1. Taunt.
> 
>     2. Shoulder Id cannot fight.
> 
>     3. When Shoulder Id would deal damage, steal 1 Æmber instead.
> 
>     4. At the start of your turn, deal 2 to each of this creature's neighbors.
> 
> 
> At the start of the Shoulder Id's controller's turn, Ability 4 resolves and Shoulder Id would deal 2 to each of the creature's neighbors. However, since Shoulder Id would deal damage, that dealing of damage is replaced with a steal via Ability 3, and Shoulder Id's controller steals 1 Æmber.
> Since all damage is dealt simultaneously, the dealing of the damage only happens once, which means the replacement of that damage will also only happen once, resulting in Shoulder Id stealing 1 Æmber.

Pretty sure it's this. I definitely agree.

Hence, there's still an issue since Shoulder ID didn't let you steal anything. Right? @gian238

**Comment by romanogit** [comment]
This one is tricky... The way it is implemented, the damage starts, while it should not.

**Comment by Osarek** [comment]
What about 2 static charge on shoulder ?
Should it steal 4 a the start of the turn?

Here is the deck of a friend with shoulder and 2 static charge

https://decksofkeyforge.com/decks/45c50a7f-631e-42a0-860f-297aa65054fd

**Comment by blinkingline** [comment]
> What about 2 static charge on shoulder ?
> Should it steal 4 a the start of the turn?

Shoulder Id would steal 2 Æ in this case.

Abilities that happen at the same timing point resolve one at a time in the order of the Active Player's choosing.  
If a Shoulder Id has two copies of Static Charge attached to it then the creature would resolve the first instance of the ability, dealing 2 damage to each neighbor. This dealing of damage is replaced by Shoulder Id's replacement effect, and its controller steals 1 Æmber. 

Then the second instance of the ability granted by Static Charge would resolve, and Shoulder Id  would do 2 damage to each of its neighbors, but the replacement effect replaces the dealing of damage with a steal, and 1 Æmber is stolen.

**Comment by OptimalDuck** [comment]
It is currently stealing two when it is on a shoulder id with two neighbours (as above it should just be one)
